docs/design/architecture.md

```diff
@@ -2,128 +2,353 @@
+
+| Process | Environment | Notes |
+|-|-|-|
+| [Agent](#agent) | VM root | Runs as the init daemon (PID 1) |
+| container workload | VM container | Managed by the agent |
+
+See also the [process overview](#process-overview).
+
+## Hypervisor
 
-The only service running in the context of the initrd is the [Agent](#agent) as the init daemon. The real workload the user wishes to run is created using libcontainer, creating a container in the same manner that is done by `runc`.
+The [configured](#configuration) [hypervisor](../hypervisors.md) creates
+a VM to host the [agent](#agent) and the [workload](#workload).
+
+> **Note:**
+>
+> The hypervisor process run inside a separate "partial container"
+> environment comprised of cgroups and a separate network namespace
+> from the host.
 
 ## Agent
 
-[`kata-agent`](../../src/agent) is a process running in the guest as a supervisor for managing containers and processes running within those containers.
+The Kata Containers agent ([`kata-agent`](../../src/agent)), written
+in the [Rust programming language](https://www.rust-lang.org), is a
+long running process that runs inside the VM. It acts as the
+supervisor for managing the containers and the [workload](#workload)
+running within those containers.
+
+> **Note:**
+>
+> Only a single agent process is run for each VM created.
+
+### Agent communications protocol
+
+The agent communicates with the other Kata components using a
+[`ttRPC`](https://github.com/containerd/ttrpc-rust) based
+[protocol](../../src/agent/protocols/protos).
+
+See also the [agent control tool](#agent-control-tool).
+
+## Workload
+
+The workload is the OCI bundle rootfs. It is exported from the host
+into the VM.
 
-For the 2.0 release, the `kata-agent` is rewritten in the [RUST programming language](https://www.rust-lang.org/) so that we can minimize its memory footprint while keeping the memory safety of the original GO version of [`kata-agent` used in Kata Container 1.x](https://github.com/kata-containers/agent). This memory footprint reduction is pretty impressive, from tens of megabytes down to less than 100 kilobytes, enabling Kata Containers in more use cases like functional computing and edge computing.
+### Workload root filesystem
 
-The `kata-agent` execution unit is the sandbox. A `kata-agent` sandbox is a container sandbox defined by a set of namespaces (NS, UTS, IPC and PID). `shimv2` can
-run several containers per VM to support container engines that require multiple
-containers running inside a pod.
+For details of how the [runtime](#runtime) make the workload image
+chosen by the user (such as `busybox:latest`) available to the
+workload process, see the [storage](#storage) section.
 
-`kata-agent` communicates with the other Kata components over `ttRPC`.
+The container workload process defined by the OCI bundle's
+configuration file (`config.json`) is created using the Kata
+Containers [agent](#agent). The agent creates a container environment
```

Comment:
It would help to remind people this is in a VM.   "The agent inside the VM creates ..."